gix/src/repository/remote.rs

```diff
@@ -65,6 +66,38 @@ impl crate::Repository {
         self.try_find_remote_inner(name_or_url, true)
     }
 
+    /// This method emulate what `git fetch <remote>` does in order to obtain a remote to fetch from.
+    ///
+    /// As such, with `name_or_url` being `Some`, it will:
+    ///
+    /// * use `name_or_url` verbatim if it is a URL, creating a new remote in memory as needed.
+    /// * find the named remote if `name_or_url` is a remote name
+    ///
+    /// If `name_or_url` is `None`:
+    ///
+    /// * use the current `HEAD` branch to find a configured remote
+    /// * fall back to either a generally configured remote or the only configured remote.
+    ///
+    /// Fail if no remote could be found despite all of the above.
+    pub fn find_fetch_remote(&self, name_or_url: Option<&BStr>) -> Result<Remote<'_>, find::for_fetch::Error> {
+        Ok(match name_or_url {
+            Some(name) => {
+                if name.contains(&b'/') || name.contains(&b'.') {
```

Comment:
I think checking for a `:` may be beneficial as well? Otherwise `server:repo` is used as remote name. Also, wouldn't it make more sense to first check whether we find a remote by name and only if not try with a url - or are we constrained by Gits implementation in that regard?

Reply:
That's all valid points, thanks for chiming in. The truth is, I didn't check against `git` in terms of code, just in terms of behaviour, so this implementation probably isn't according to 'spec'. Now that it moved 'up' from `gitoxide-core` to `gix` it should be according to spec though.

I will take another look and see what needs to change to be what `git` does.

Reply:
It looks like [this is how git decides how to interpret the `name_or_url`](https://github.com/git/git/blob/6807fcfedab84bc8cd0fbf721bc13c4e68cda9ae/remote.c#L673-L701). Some minor modifications can be made easily, but I also noticed that there are so called remotes-files and branches-files (!!) - yet another way to configure things apparently, and something I have never encountered or heard off. 

Ignoring that, I have just modified the detection routine to be in-line with what `git` does, and it happened to have been quite exactly what you suggested :). Thanks again!

Reply:
> I also noticed that there are so called remotes-files and branches-files

I never had of those either, the amount of features Git has really is crazy. Anyways, glad I could help :)